lib/wallet/src/Cardano/Wallet.hs

```diff
@@ -2534,7 +2534,7 @@ constructTransaction
     -> WalletId
     -> Cardano.AnyCardanoEra
     -> TransactionCtx
-    -> SelectionOf TxOut
+    -> Either PreSelection (SelectionOf TxOut)
```

Comment:
Hi @paweljakubas 

I'm a bit confused about why this argument needs to be of type `Either PreSelection Selection`.

Is there a good reason for this? (Perhaps preparing for the future?)

AFAICT, this function is only ever called with `Left PreSelection`.

See the following commit, which still compiles: https://github.com/input-output-hk/cardano-wallet/pull/3567/commits/262b7b939ae9d64c242f50410b1b3209858397f2.

Reply:
hi @jonathanknowles indeed. `PreSelection` was introduced as the intention was to construct tx WITHOUT inputs and `SelectionOf` has `NonEmpty` of inputs. But when I proceeded then it turn out that node does not accept constructing tx with empty inputs. Hence this `dummyInput` hack. But, node intends to lift this restriction in coming version of node, so I intended to stick to differentiating `PreSelection` and `SelectionOf`.